### PR TITLE
checker: add checks for map key/value types

### PR DESCRIPTION
**Pull request by Noctember**
Before, this would make a C error. This adds checks to make sure both types exists
```
struct App {
	my_map map[string]DoesNotExist
}
```

### CODE REVIEW

**Comment by medvednikov** [comment]
Thanks! Can you add a small checker test for this error?

**Comment by Noctember** [comment]
Done 👍

File: vlib/v/checker/tests/map_unknown_type.out [hunk 1]
```diff
@@ -0,0 +1,5 @@
+vlib/v/checker/tests/map_unkown_value.vv:2:5: error: unknown type `DoesNotExist`
+    1 | struct App {
+    2 |     my_map map[string]DoesNotExist
+      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    3 | }
```
Comment (Delta456) [inline vlib/v/checker/tests/map_unknown_type.out:5]: Please add a newline here

File: vlib/v/checker/tests/map_unknown_value.vv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+struct App {
+    my_map map[string]DoesNotExist
+}
```
Comment (Delta456) [inline vlib/v/checker/tests/map_unknown_value.vv:3]: Same here

**Comment by medvednikov** [comment]
`V panic: failed to open file "vlib/v/checker/tests/map_unknown_value.out"`

**Comment by medvednikov** [comment]
@Noctember 

```
expected:
vlib/v/checker/tests/map_unkown_value.vv:2:5: error: unknown type `DoesNotExist`
    1 | struct App {
    2 |     my_map map[string]DoesNotExist
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    3 | }
============
found:

============
```

**Comment by Noctember** [comment]
Oh... thats awkward

**Pull request merged**